scanpy/plotting/_tools/scatterplots.py

```diff
@@ -434,7 +436,7 @@ def embedding(
             path_effect = None
 
         # Adding legends
-        if categorical:
+        if categorical or color_vector.dtype == bool:  # for categorical and boolean
```

Comment:
```suggestion
        if categorical or color_vector.dtype == bool:
```

That's what the code already says. Adding some reasoning above would be more interested. But think this is fine without any comments.